### ISSUE TITLE
Proposal: Date support

### ISSUE THREAD

**Issue by 4F2E4A2E** [19majkel94/type-graphql#22]
I propose to add Date support using this [implementation](https://gist.github.com/langpavel/b30f3d507a47713b0c6e89016e4e9eb7), found here https://github.com/graphql/graphql-js/issues/550

**Comment by 19majkel94** [comment]
Honestly, I prefer the ISO version of the Date scala, it's human readable and easier to debug:
```ts
import { GraphQLScalarType, Kind } from "graphql";

export const DateScalar = new GraphQLScalarType({
  name: "Date",
  description: "Date scalar type",
  parseValue(value: string) {
    return new Date(value); // value from the client input variables
  },
  serialize(value: Date) {
    return value.toISOString(); // value sent to the client
  },
  parseLiteral(ast) {
    if (ast.kind === Kind.STRING) {
      return new Date(ast.value); // value from the client query
    }
    return null;
  },
});
```
And because I don't want to make this framework so opinionated, I prefer to gave developers freedom to  use their own version of `Date` and other scalars with #5 custom scalar support 😉

**Comment by 4F2E4A2E** [comment]
You are right.  I will change it to the iso format.

**Comment by 4F2E4A2E** [comment]
A default date speeds up the getting started, but of course, passing in the custom scalar type as presented by you is just fine too:
```typescript
import { GraphQLScalarType, Kind } from 'graphql';
import { Field, GraphQLObjectType } from 'type-graphql';

export const DateScalar = new GraphQLScalarType({
  name:        'Date',
  description: 'Date scalar type',
  parseValue(value: string) {
    return new Date(value); // value from the client input variables
  },
  serialize(value: Date) {
    return value.toISOString(); // value sent to the client
  },
  parseLiteral(ast) {
    if (ast.kind === Kind.STRING) {
      return new Date(ast.value); // value from the client query
    }
    return null;
  }
});

@GraphQLObjectType()
export class User {
  @Field(type => DateScalar)
  contributorsince: Date;
  @Field()
  alive: boolean;
}
```

**Comment by 19majkel94** [comment]
Closing - implemented in ba21848 😉
